File: frontend/src/models/representation/RepresentationFactory.ts
import Representation, {RepresentationFile} from "./Representation";
import AccessesRepresentation from "./files/AccessesRepresentation";
import IDToEntityRepresentation from "./files/IDToEntityRepresentation";
import AuthorRepresentation from "./files/AuthorRepresentation";
import CommitRepresentation from "./files/CommitRepresentation";
import EntityToIDRepresentation from "./files/EntityToIDRepresentation";
import CodeEmbeddingsRepresentation from "./files/CodeEmbeddingsRepresentation";

export abstract class RepresentationFactory {
    static getRepresentation(representation: any) : Representation {
        switch (representation.type) {
            case RepresentationFile.ACCESSES:
                return new AccessesRepresentation(representation);
            case RepresentationFile.IDTOENTITIY:
                return new IDToEntityRepresentation(representation);
            case RepresentationFile.AUTHOR:
                return new AuthorRepresentation(representation);
            case RepresentationFile.COMMIT:
                return new CommitRepresentation(representation);
            case RepresentationFile.ENTITY_TO_ID:
                return new EntityToIDRepresentation(representation);
            case RepresentationFile.CODE_EMBEDDINGS:
                return new CodeEmbeddingsRepresentation(representation);
            default:
                throw new Error('Type ' + representation.type + ' unknown.');
        }
    }

    static getRepresentations(representation: any[]) : Representation[] | null {
        if (representation === null)
            return null;
        return representation.map((representation:any) => this.getRepresentation(representation))
    }
}

File: backend/src/main/java/pt/ist/socialsoftware/mono2micro/representation/domain/RepresentationFactory.java
package pt.ist.socialsoftware.mono2micro.representation.domain;

import static pt.ist.socialsoftware.mono2micro.representation.domain.AccessesRepresentation.ACCESSES;
import static pt.ist.socialsoftware.mono2micro.representation.domain.AuthorRepresentation.AUTHOR;
import static pt.ist.socialsoftware.mono2micro.representation.domain.CodeEmbeddingsRepresentation.CODE_EMBEDDINGS;
import static pt.ist.socialsoftware.mono2micro.representation.domain.CommitRepresentation.COMMIT;
import static pt.ist.socialsoftware.mono2micro.representation.domain.EntityToIDRepresentation.ENTITY_TO_ID;
import static pt.ist.socialsoftware.mono2micro.representation.domain.IDToEntityRepresentation.ID_TO_ENTITY;

public class RepresentationFactory {
    public static Representation getRepresentation(String representationType) {
        switch (representationType) {
            case ACCESSES:
                return new AccessesRepresentation();
            case ID_TO_ENTITY:
                return new IDToEntityRepresentation();
            case AUTHOR:
                return new AuthorRepresentation();
            case COMMIT:
                return new CommitRepresentation();
            case ENTITY_TO_ID:
                return new EntityToIDRepresentation();
            case CODE_EMBEDDINGS:
                return new CodeEmbeddingsRepresentation();
            default:
                throw new RuntimeException("The type \"" + representationType + "\" is not a valid representation type.");
        }
    }
}
